app/workers/certificates.php

```diff
@@ -26,169 +28,224 @@ public function init(): void
+                throw new ExceptionCertificate('Failed to rename certificate chain.pem: ' . \json_encode($stdout), true);
+            }
+    
+            if (!@\rename('/etc/letsencrypt/live/' . $domain->get() . '/fullchain.pem', APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/fullchain.pem')) {
+                throw new ExceptionCertificate('Failed to rename certificate fullchain.pem: ' . \json_encode($stdout), true);
+            }
+    
+            if (!@\rename('/etc/letsencrypt/live/' . $domain->get() . '/privkey.pem', APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/privkey.pem')) {
+                throw new ExceptionCertificate('Failed to rename certificate privkey.pem: ' . \json_encode($stdout), true);
+            }
 
-            $certificate = $dbForConsole->updateDocument('domains', $certificate->getId(), $certificate);
-
-            if(!$certificate) {
-                throw new Exception('Failed saving domain to DB');
+            // This multi-line syntax helps IDE
+            $config =
+                "tls:" .
+                "  certificates:" . 
+                "    - certFile: /storage/certificates/{$domain->get()}/fullchain.pem" . 
+                "      keyFile: /storage/certificates/{$domain->get()}/privkey.pem";
+            
+            // Save configuration into Traefik using our new cert files
+            if (!\file_put_contents(APP_STORAGE_CONFIG . '/' . $domain->get() . '.yml', $config)) {
+                throw new ExceptionCertificate('Failed to save Traefik configuration.', true);
             }
-        }
 
-        $config =
-"tls:
-  certificates:
-    - certFile: /storage/certificates/{$domain->get()}/fullchain.pem
-      keyFile: /storage/certificates/{$domain->get()}/privkey.pem";
+            // Read new renew date from cert file
+            // TODO: This might not be required, we could calculate it. But this feels safer
+            $certPath = APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/cert.pem';
+            $certData = openssl_x509_parse(file_get_contents($certPath));
+            $validTo = $certData['validTo_time_t'];
+            $expiryInAdvance = (60*60*24*30);
+            $certificate->setAttribute('renewDate', $validTo - $expiryInAdvance);
+
+            // All went well at this point 🥳
+            
+            // Reset attempts count for next renwal
+            $certificate->setAttribute('attempts', 0);
+
+            // Mark issue date
+            $certificate->setAttribute('issueDate', \time());
+        } catch(ExceptionCertificate $e) {
+            // These exceptions are expected if renew shouldn't or can't happen
+
+            // Add exception as log into certificate
+            $certificate->setAttribute('log', $e->getMessage());
+
+            $attempt = $certificate->getAttribute('attempts', 0);
+            $attempt++;
+            
+            // Save increased attempts count if requested by exception
+            if($e->getIncrementAttempts()) {
```

Comment:
As mentioned in the comment above, we shouldn't rely on the exception to track # of failed attempts.

Reply:
We no longer take this from exception, we now increment count with all failed attempts.